test/base.service.test.js

```diff
@@ -0,0 +1,71 @@
+require('../src/util/dbconfig');
+const testMeasure = require('./fixtures/testMeasure.json');
+const testLibrary = require('./fixtures/testLibrary.json');
+const testPatient = require('./fixtures/testPatient.json');
+const { testSetup, cleanUpDb } = require('./populateTestData');
+const supertest = require('supertest');
+const { buildConfig } = require('../src/util/config');
+const { initialize } = require('../src/server/server');
+const config = buildConfig();
+const server = initialize(config);
+const updatePatient = { id: 'testPatient', name: 'anUpdate' };
+describe('base.service', () => {
+  beforeAll(async () => {
+    await testSetup(testMeasure, testPatient, testLibrary);
+  });
+
+  describe('create', () => {
+    test('test create with correct headers', async () => {
+      await supertest(server.app)
+        .post('/4_0_0/Patient')
+        .send(testPatient)
+        .set('Accept', 'application/json+fhir')
+        .set('content-type', 'application/json+fhir')
+        .expect(201)
+        .then(async response => {
+          // Check the response
+          expect(response.headers.location).toBeDefined();
+        });
+    });
+  });
+  describe('searchById', () => {
+    test('test searchById with correctHeaders and  the id should be in database', async () => {
+      await supertest(server.app)
+        .get('/4_0_0/Patient/testPatient')
+        .set('Accept', 'application/json+fhir')
+        .expect(200)
+        .then(async response => {
+          expect(response.body.id).toEqual(testPatient.id);
+        });
+    });
+  });
+  describe('update', () => {
+    //*a put request*/
+    test('test update with correctHeaders and  the id is in database', async () => {
+      await supertest(server.app)
+        .put('/4_0_0/Patient/testPatient')
+        .send(updatePatient)
+        .set('Accept', 'application/json+fhir')
+        .set('content-type', 'application/json+fhir')
+        .expect(200)
+        .then(async response => {
+          // Check the response
+          expect(response.body.location).not.toBeNull();
```

Comment:
```suggestion
          expect(response.headers.location).not.toBeNull();
```
response.body.location is undefined which is not null, so the test passes, but I imagine we want the response.headers.location here